src/components/tl/ucp/allgather/allgather_knomial.c

```diff
@@ -191,16 +253,31 @@ ucc_status_t ucc_tl_ucp_allgather_knomial_init_r(
     ucc_rank_t         size    = UCC_TL_TEAM_SIZE(tl_team);
     ucc_rank_t         broot   = 0;
     ucc_tl_ucp_task_t *task;
+    ucc_knomial_pattern_t p;
 
-    if (coll_args->args.coll_type == UCC_COLL_TYPE_BCAST) {
-        broot = coll_args->args.root;
-        rank = VRANK(rank, broot, size);
+    if (coll_args->args.coll_type == UCC_COLL_TYPE_ALLGATHER) {
+        ucc_kn_ag_pattern_init(size, rank, radix,
```

Comment:
looks you are right - not needed here. copy-pasted from reduce-scatter knomial. there pattern init is called there because i need to understand which ranks are extra/proxy - this info is part of kn pattern